index.js

```diff
@@ -551,6 +551,6 @@ function createError (code, message, statusCode) {
 }
 
 module.exports = fp(compressPlugin, {
-  fastify: '3.x',
+  fastify: '4.x',
```

Comment:
I think here we can keep `≥3`, as the plugin code doesn't change. It only changes how plugin users are registering it. But this code *as is* could still work in v3.x
```suggestion
  fastify: '>=3.x',
```

Reply:
From past experiences, maintaining that is a nightmare. We need separate tests for v3 and v4. I prefer to bump the major so we can avoid any compatibility issues down the line.